os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_central/ble_tizenrt_central_client_app.c

```diff
@@ -181,8 +181,10 @@ void ble_tizenrt_central_handle_callback_msg(T_TIZENRT_APP_CALLBACK_MSG callback
             if(ble_tizenrt_write_no_rsp_sem != NULL)
                 os_mutex_give(ble_tizenrt_write_no_rsp_sem);
 
-            trble_conn_handle disconnected = (uint32_t) callback_msg.u.buf;
-            client_init_parm->trble_device_disconnected_cb(disconnected);
+            uint32_t temp = (uint32_t) callback_msg.u.buf;
+            uint8_t reason = temp & 0xff;
+            trble_conn_handle disconnected = temp >> 16;
```

Comment:
Can you please add a comment about why you are shifting by 16 bits?

Reply:
'ble msg handler' can send only 1 pointer as a parameter. So to send 2 params, I combined them by bit shifting.
The total of 'conn_id' and 'disc_cause' is 24 bits, so I used 32 bits variable.
The first 16-bits are for conn_id, rest of bits are for disc_cause.